src/textual/command.py

```diff
@@ -504,11 +513,21 @@ def on_mount(self, _: Mount) -> None:
         )
 
         assert self._calling_screen is not None
-        self._sources = [
-            source(self._calling_screen, match_style) for source in self._source_classes
+        self._providers = [
+            provider_class(self._calling_screen, match_style)
+            for provider_class in self._provider_classes
         ]
-        for _source in self._sources:
-            _source._post_init()
+        for provider in self._providers:
+            provider._post_init()
+
+    async def on_unmount(self) -> None:
```

Comment:
There's an asymmetry with how exceptions in `shutdown` are handled, vs `startup` (the latter reporting as errors in `search` does too). An error in `startup` is nicely reported in the console:

![Screenshot 2023-09-14 at 11 32 18](https://github.com/Textualize/textual/assets/28237/2e926a6b-ca93-4cf8-ac25-78950c49e383)

whereas an error in `showdown` is like this:

![Screenshot 2023-09-14 at 11 32 35](https://github.com/Textualize/textual/assets/28237/7c2c375c-2dec-4384-9695-b75042e04351)